### PR TITLE
Submitting response

### PR DESCRIPTION
**Pull request by ayushdevelop**
# Related Issue

- https://github.com/code-mancers/g-bot/issues/4

# Functionalities Added or Changed

- On clicking the `Submit` button visible on the modal the response will be populated in the channel set for exampe `gather-bot`
- The response will be submitted as user

# Checklist

- [x] Uploaded screenshots

# Screenshots
Response of Standup

### CODE REVIEW

File: server/src/constants/standupQuestions.ts [hunk 1]
```diff
@@ -3,20 +3,20 @@ type Questions = {
   id: string;
 };
 const questions: Questions[] = [
-  { question: "How do you feel today?", id: "question_one" },
+  { question: 'How do you feel today?', id: 'question_one' },
```
Comment (SujayPrabhu96) [inline server/src/constants/standupQuestions.ts:6]: When prettier configuration is to use double quotes why this change included ?

File: server/src/listeners/actions/notWorking.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+type NotWorkingArguments = {
```
Comment (SujayPrabhu96) [inline server/src/listeners/actions/notWorking.ts:1]: Not working feature is added in next PR right ? Why it is in this PR ?

File: server/src/listeners/views/modalView.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
+type StandupModalArguments = {
+  client: {
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:2]: As the arguments type remains same for actions, views, we can define the type in `src/types` and import it wherever required

File: server/src/listeners/views/modalView.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  body: { user: { id: string } };
+};
+
+const standupModal = async ({
+  ack,
+  client,
+  payload,
+  context,
+  body,
+}: StandupModalArguments) => {
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:19]: Optional: As in frontend we usually name it `ComponentProps`
```suggestion
}: StandupModalArgs) => {
```

File: server/src/listeners/views/modalView.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      (data: { profile: { image_192: string; diplay_name: string } }) =>
+        data.profile
+    );
+
+  try {
+    await client.chat.postMessage({
+      //posting message as a user
+      icon_url: user.image_192,
+      username: user.display_name,
+
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:41]: remove empty line

File: server/src/listeners/views/modalView.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      token: context.botToken,
+
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:43]: same

File: server/src/listeners/views/modalView.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      // Channel to send message to
+      channel: "C0403497QV8",
+
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:46]: same^

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: server/src/listeners/views/modalView.ts [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+            },
+          ],
+        },
+      ],
+    });
+  } catch (error) {
+    await client.chat.postMessage({
+      token: context.botToken,
+      // Channel to send message to
+      channel: "C0403497QV8",
```
Comment (SujayPrabhu96) [inline server/src/listeners/views/modalView.ts:114]: Even though this hard-coding is for temporary, I would suggest to put it in env

Comment (ayushdevelop) [reply server/src/listeners/views/modalView.ts:114]: okay doing it

**Review by SujayPrabhu96 — APPROVED** [review]

**Pull request merged**